### DISCUSSION TITLE
How to use environment variable, ANDROID_PUBLISHER_CREDENTIALS

### DISCUSSION THREAD

**Discussion by ManisDroid** [Triple-T/gradle-play-publisher#943] [Q&A] [answered]
We have a question related to Authenticating Gradle Play Publisher section

This is the first time, we are integrating GPP for publishing in the PlayStore. 
We can make it successful to publish the appBundle in the PlayStore using serviceAccountCredentials.set(file("your-key.json")). As a security issue, we like to use environment variable as suggested by the library.

So, we set the ANDROID_PUBLISHER_CREDENTIALS environment variable as below

export ANDROID_PUBLISHER_CREDENTIALS = path to the json file.

But when we execute, ./gradlew bootstrap

we got this error
A failure occurred while executing com.github.triplet.gradle.play.tasks.Bootstrap$DetailsDownloader
      > Expected value at line 1 column 1 path $

Can you please guide us, to fix this issue? 

Kind Regards
Mani

**Comment by SUPERCILEX** [comment, marked as answer]
No, the environment variable has to contain the contents of the credentials (i.e. the contents of your file). Though if you have a file available, why not simply use that instead of `file("your-key.json")`?

**Reply by ManisDroid** [reply]
Thanks for a quick reply. I understand the first part ie., to use the content of the file to set it in the environment variable.

Sorry, I dont understand this statement, 'Though if you have a file available, why not simply use that instead of file("your-key.json")'

Thanks for your time.

**Reply by SUPERCILEX** [reply]
Well you wanted to put a file path in the environment variable. That means you have a file available somewhere right? So just do `serviceAccountCredentials.set(file("path/to/the/json/file"))`

**Reply by ManisDroid** [reply]
Thanks a lot. I did it. It works now. Thanks for timely reply. Have a good day.
